src/NServiceBus.AmazonSQS/MessagePump.cs

```diff
@@ -321,7 +322,7 @@ async Task ProcessMessage(Message receivedMessage, CancellationToken token)
                     isPoisonMessage = true;
                 }
 
-                if (incomingMessage == null || transportMessage == null)
+                if (messageBody == null || transportMessage == null)
                 {
                     Logger.Warn($"Treating message with SQS Message Id {receivedMessage.MessageId} as a poison message because it could not be converted to an IncomingMessage. Moving to error queue.");
```

Comment:
`IncomingMessage` is not anymore a thing, the log message should be changed.

Reply:
My brain read this is incoming message. Good catch